server/src/main/java/org/jboss/as/server/deployment/module/descriptor/DeploymentStructureDescriptorParser.java

```diff
@@ -238,9 +240,25 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
         }
     }
 
-    private void handleDeployment(final DeploymentPhaseContext phaseContext, final DeploymentUnit deploymentUnit, final ModuleSpecification moduleSpec, final ModuleStructureSpec rootDeploymentSpecification, Set<ModuleIdentifier> additionalModules) throws DeploymentUnitProcessingException {
+    private void handleDeployment(final DeploymentPhaseContext phaseContext, final DeploymentUnit deploymentUnit, final ModuleSpecification moduleSpec, final ModuleStructureSpec rootDeploymentSpecification, final Set<AdditionalModuleSpecification> additionalModules) throws DeploymentUnitProcessingException {
         final Map<VirtualFile, ResourceRoot> resourceRoots = resourceRoots(deploymentUnit);
-        moduleSpec.addUserDependencies(rootDeploymentSpecification.getModuleDependencies());
+        // handle unmodifiable module dependencies with alias
+        List<ModuleDependency> moduleDependencies = rootDeploymentSpecification.getModuleDependencies();
+        List<ModuleDependency> copy = new ArrayList<>();
+        List<ModuleDependency> aliasDependencies = new ArrayList<>();
+        copy.addAll(moduleDependencies);
+        Iterator<ModuleDependency> it = copy.iterator();
```

Comment:
Minor comment, but this could be a bit more readable as a for-each.

Reply:
changed to use for-each.